sale_order_import/wizard/import_sale.py

```diff
@@ -2,202 +2,480 @@
+            error_line_vals['error'] = True
+        else:
+            self._get_picking_dict(warehouse_value, picking_dict,
+                                   warehouse_dict, error_line_vals)
+
+        team_value = row[team_id].strip()
+        if not team_value:
+            error_line_vals['error_name'] = error_line_vals['error_name']\
+                + _('Team is empty!') + '\n'
+            error_line_vals['error'] = True
+        else:
+            self._get_team_dict(team_value, team_dict, error_line_vals)
+
+        carrier_value = row[carrier_id].strip()
+        if not carrier_value:
+            error_line_vals['error_name'] = error_line_vals['error_name']\
+                + _('Carrier is empty!') + '\n'
+            error_line_vals['error'] = True
+        else:
+            self._get_carrier_dict(carrier_value, carrier_dict,
+                                   error_line_vals)
+
+        return partner_value, product_id_value, pricelist_value,\
+            warehouse_value, team_value, carrier_value
+
+    @api.model
+    def _get_order_value(self, row, error_line_vals, taxes, price_unit,
+                         taxes_id, product_qty):
+        """Get order value"""
+        tax_from_chunk = row[taxes_id].strip()
+        if tax_from_chunk:
+            self._get_taxes(tax_from_chunk, taxes, error_line_vals)
+
+        qty = row[product_qty].strip()
+        if not qty:
+            error_line_vals['error_name'] = error_line_vals['error_name']\
+                + _('Quantity is empty!') + '\n'
+            error_line_vals['error'] = True
+        else:
+            qty = float(qty)
+
+        if qty <= 0:
+            error_line_vals['error_name'] = error_line_vals['error_name']\
+                + _('Quantity not less than zero!') + '\n'
+            error_line_vals['error'] = True
+
+        price_unit_value = row[price_unit].strip()
+        if not price_unit_value:
+            error_line_vals['error_name'] = error_line_vals['error_name']\
+                + _('Unit Price is empty!') + '\n'
+            error_line_vals['error'] = True
+        else:
+            price_unit_value = float(price_unit_value)
+
+        if price_unit_value <= 0:
+            error_line_vals['error_name'] = error_line_vals['error_name']\
+                + _('Unit Price not less than zero!') + '\n'
+            error_line_vals['error'] = True
+        return qty, price_unit_value
+
+    @api.model
+    def _get_order_item_dict(self, error_log_id, row, order, taxes, line_name,
+                             product_dict, product_id_value, order_item_dict,
+                             qty, price_unit_value):
+        """Get order item dict"""
+        if not error_log_id:
+            name = row[line_name].strip()
+            product_data = self.env['product.product'].browse(
+                product_dict[product_id_value]
+            )  # odoo11
+            if not name:
+                name = product_data['value']['name']
```

Comment:
If the "Line Description" of the CSV file is empty, Odoo Server Error will result,
```
Traceback (most recent call last):
  File "/home/tslai/ws/sst/odoo/odoo/http.py", line 646, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/home/tslai/ws/sst/odoo/odoo/http.py", line 307, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/home/tslai/ws/sst/odoo/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/home/tslai/ws/sst/odoo/odoo/http.py", line 683, in dispatch
    result = self._call_function(**self.params)
  File "/home/tslai/ws/sst/odoo/odoo/http.py", line 339, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/home/tslai/ws/sst/odoo/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/home/tslai/ws/sst/odoo/odoo/http.py", line 332, in checked_call
    result = self.endpoint(*a, **kw)
  File "/home/tslai/ws/sst/odoo/odoo/http.py", line 927, in __call__
    return self.method(*args, **kw)
  File "/home/tslai/ws/sst/odoo/odoo/http.py", line 512, in response_wrap
    response = f(*args, **kw)
  File "/home/tslai/ws/sst/odoo/addons/web/controllers/main.py", line 928, in call_button
    action = self._call_kw(model, method, args, {})
  File "/home/tslai/ws/sst/odoo/addons/web/controllers/main.py", line 916, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/home/tslai/ws/sst/odoo/odoo/api.py", line 689, in call_kw
    return call_kw_multi(method, model, args, kwargs)
  File "/home/tslai/ws/sst/odoo/odoo/api.py", line 680, in call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/home/tslai/ws/sst/sts-custom/sale_order_import/wizard/import_sale.py", line 645, in import_sale_data
    qty, price_unit_value)
  File "/home/tslai/ws/sst/sts-custom/sale_order_import/wizard/import_sale.py", line 203, in _get_order_item_dict
    name = product_data['value']['name']
  File "/home/tslai/ws/sst/odoo/odoo/models.py", line 4657, in __getitem__
    return self._fields[key].__get__(self, type(self))
KeyError: 'value'
```